src/core/meshcop/border_agent_txt_data.hpp

```diff
@@ -258,6 +258,20 @@ class TxtData : public InstanceLocator
 #endif
     };
 
+    /**
+     * Defines experimental TXT keys.
+     *
+     * Experimental keys are prefixed with an underscore `_`. They are for testing and development purpose and may be
+     * changed or removed in the future.
+     *
+     */
+    struct KeyExperimental
```

Comment:
I think we should simply add this in existing `struct Key`. No need to define a new `struct KeyExperimental`.